backend/src/resolvers/comments.js

```diff
@@ -2,44 +2,47 @@ import { neo4jgraphql } from 'neo4j-graphql-js'
 import { UserInputError } from 'apollo-server'
 
 const COMMENT_MIN_LENGTH = 1
-export default {
-  Query: {
-    CommentByPost: async (object, params, context, resolveInfo) => {
-      const { postId } = params
-
-      const session = context.driver.session()
-      const transactionRes = await session.run(`
-          MATCH (comment:Comment)-[:COMMENTS]->(post:Post {id: $postId})
-          RETURN comment {.id, .contentExcerpt, .createdAt} ORDER BY comment.createdAt ASC`, {
-        postId
-      })
+const NO_POST_ERR_MESSAGE = 'Comment cannot be created without a post!'
 
-      session.close()
-      let comments = []
-      transactionRes.records.map(record => {
-        comments.push(record.get('comment'))
-      })
-
-      return comments
-    }
-  },
+export default {
   Mutation: {
     CreateComment: async (object, params, context, resolveInfo) => {
       const content = params.content.replace(/<(?:.|\n)*?>/gm, '').trim()
+      const { postId } = params
+      // Adding relationship from comment to post by passing in the postId,
+      // but we do not want to create the comment with postId as an attribute
+      // because we use relationships for this. So, we are deleting it from params
+      // before comment creation.
+      delete params.postId
 
       if (!params.content || content.length < COMMENT_MIN_LENGTH) {
         throw new UserInputError(`Comment must be at least ${COMMENT_MIN_LENGTH} character long!`)
       }
-      const { postId } = params
-      delete params.postId
-      const comment = await neo4jgraphql(object, params, context, resolveInfo, false)
+      if (!postId.trim()) {
+        throw new UserInputError(NO_POST_ERR_MESSAGE)
+      }
 
       const session = context.driver.session()
+      const postQueryRes = await session.run(`
+        MATCH (post:Post {id: $postId})
+        RETURN post`, {
+        postId
+      }
+      )
+      const [post] = postQueryRes.records.map(record => {
+        return record.get('post')
+      })
+
+      if (!post) {
+        throw new UserInputError(NO_POST_ERR_MESSAGE)
+      }
+      const comment = await neo4jgraphql(object, params, context, resolveInfo, false)
 
       await session.run(`
-          MATCH (post:Post {id: $postId}), (comment:Comment {id: $commentId})
-          MERGE (post)<-[:COMMENTS]-(comment)
-          RETURN comment {.id, .content}`, {
+        MATCH (post:Post {id: $postId}), (comment:Comment {id: $commentId}), (author:User {id: $userId})
```

Comment:
> [<img alt="Tirokk" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/Tirokk) **Authored by [Tirokk](https://github.com/Tirokk)**
_<time datetime="2019-05-04T09:06:59Z" title="Saturday, May 4th 2019, 11:06:59 am +02:00">May 4, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -31,15 +11,15 @@ export default {
         throw new UserInputError(`Comment must be at least ${COMMENT_MIN_LENGTH} character long!`)
       }
       const { postId } = params
-      delete params.postId
       const comment = await neo4jgraphql(object, params, context, resolveInfo, false)
 
       const session = context.driver.session()
 
       await session.run(`
-          MATCH (post:Post {id: $postId}), (comment:Comment {id: $commentId})
-          MERGE (post)<-[:COMMENTS]-(comment)
-          RETURN comment {.id, .content}`, {
+        MATCH (post:Post {id: $postId}), (comment:Comment {id: $commentId}), (author:User {id: $userId})
```


I think and hope this is a good solution with a single cypher statement. @mattwr18

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-05-04T12:20:33Z" title="Saturday, May 4th 2019, 2:20:33 pm +02:00">May 4, 2019</time>_

---


I think that it is! 
I found the idea on neo4j's docs https://neo4j.com/docs/cypher-manual/current/clauses/merge/#merge-merge-on-multiple-relationships

I am very happy to have found the error in my backend test because before I was running two sessions to associate the posts and author. I think this could be a performance issue(?) 

We need to write some tests that lead to the implementation  where a comment cannot be created without a post and author.